### ISSUE TITLE
Some bosses didn't reset correctly 

### ISSUE THREAD

**Issue by Err0r1** [azerothcore/azerothcore-wotlk#7496]
### Current Behaviour

If you trigger a boss or die in a boss fight, the boss doesn't reset itself completely.

The problem affects Sindragosa (instant attack), Halion and Prof (they do not transform back).

### Expected Blizzlike Behaviour

The boss should completely reset himself with all his events/spells 

### Steps to reproduce the problem

1. Go to ICC
2. Fight Prof, die in phase 3 (after transform)
3. Go to Sindra, kill the 2 dragons, die, go to Sindra
##
1. Go to halion
2. Fight and die after phase 2

### Extra Notes

https://www.youtube.com/watch?v=no7HLQhKSGY (Credits Zalacar)

Skip to last 16 secs to see sindra attacking instantly

### AC rev. hash/commit

dd833a25a63679d20382e471ccd3ad6646672d6a

### Operating system

Debian 10

### Custom changes or Modules

AutoBalance
1v1 Arena
CFBG
Eluna Lua
SoloLFG
Transmog

**Comment by T1ti** [comment]
https://github.com/azerothcore/azerothcore-wotlk/issues/7153

**Comment by steenburgh** [comment]
Closing as a duplicate of #7153, LMK if you think this is incorrect

**Comment by steenburgh** [comment]
Also, THE ICY WINDS OF NORTHREND WILL CONSUME YOUR SOULS
